src/vm/util.cpp

```diff
@@ -1911,6 +1912,31 @@ size_t GetLargestOnDieCacheSize(BOOL bTrueSize)
                         }   
                     }
 
+                    // TODO: Currently GetLogicalCpuCountFromOS() and GetLogicalCpuCountFallback() are broken on 
```

Comment:
This whole codepath seems to be really outdated to me. At one point I asked around to see if these comments still apply, and to the knowledge of everyone I talked to the OS issues that this function is working around were fixed a long time ago.

We should probably rewrite this whole thing at some point...

Reply:
We should be using GetLogicalProcessorInformationEx on the OS versions where it's available and fall back to cpuid where it's not. but you don't have to do it in this PR; feel free to open an issue to track if you don't want to do it in this PR.

---
In reply to: [163342330](https://github.com/dotnet/coreclr/pull/15975#discussion_r163342330) [](ancestors = 163342330)

Reply:
I have created https://github.com/dotnet/coreclr/issues/16071.